include/object/structures.h

```diff
@@ -300,6 +299,11 @@ struct tcb {
     struct tcb *tcbEPNext;
     struct tcb *tcbEPPrev;
 
+#ifdef CONFIG_HAVE_FPU
+    /* Currently only used for seL4_TCBFlag_fpuDisabled */
+    seL4_TCBFlag flags;
+#endif
```

Comment:
Should we keep the field in the TCB struct in all configs instead?

It's an additional field, but it doesn't change the power of 2 of the TCB size, so shouldn't affect anything else. There are two sides for that for verification (will need small changes in otherwise unaffected verification configs, but will make everything more uniform). Raf, Corey any opinion on that?

Reply:
My first reaction is that making this uniform across configs would be nice.

I might be missing some context, but since I'm assuming that we're implementing it this way to be able to support adding other flags in the future, then it makes sense to me to do it properly now instead of needing to come back in the future if/when we add generic flags. As you said, it shouldn't add much verification work now and it might help us avoid accidentally using any arch assumptions in the proofs.

Reply:
I like uniformity. It indeed doesn't bump up TCB size, and I don't think there's much benefit to dropping it.

Reply:
If size is a problem, we can make the bool variables one byte instead of a `word_t`.

Reply:
Given the next thing in the struct is very likely going to be `word_t`-aligned, e.g. scheduling context on MCS, a byte would need padding anyway, so might as well keep `word_t`. I think there's still plenty of space in the TCB.

Reply:
Actually, we have the following 1 byte fields next to each other in `tcb_t`: `tcbDomain`, `tcbMCP`, `tcbPriority`. Making them one byte would save two words, even with padding. `tcbAffinity` could be 1, 2 or 4 bytes too, and that would save another word.

Saving 24 bytes is almost half a cache line, so it would be worth it just for the increased cache hit rate.

Also, we're quite close to the size limit on some architectures, e.g. I couldn't align `fpuState` to 64 bytes on Aarch64, only to 32 bytes.

(Real savings could be made if `cte_t` is allowed to be after `tcb_t` though, or if the unused area is replaced with explicit padding.)

Reply:
I think we'd be fine with squashing it among the 3 other 8-bit values, not so sure if this is the time to start rearranging the TCB for size savings. I'm surprised to hear we're getting close to the size limit though!